### ISSUE TITLE
Inject arbitrary secrets into the environment

### ISSUE THREAD

**Issue by alesten** [neo4j-contrib/neo4j-helm#94]
Add support for injecting secrets into the environment. Just like there is now for configmap in `core.configMap`. 

Use-case I have where this would be needed:
I have a custom plugin that exposes a procedure, that when called puts a message into a rabbitMQ queue. This procedure is called in an apoc trigger for different events. The rabbitMQ credentials is pulled from the environment. 
Right now these credentials will need to be put into a configmap in order to support the use-case, but as the credentials are sensitive, a way to take them from a secret would be better than configmap.

**Comment by moxious** [comment]
Can you please be more specific about what you'd want to inject from what secret?   As an example, right now we inject a single env var NEO4J_SECRETS_PASSWORD from a secret, like this:  

```
        env:
          {{- if .Values.authEnabled }}
          - name: NEO4J_SECRETS_PASSWORD
            valueFrom:
              secretKeyRef:
                {{- if .Values.existingPasswordSecret }}
                name: {{ .Values.existingPasswordSecret }}
                {{- else }}
                name: {{ template "neo4j.secrets.fullname" . }}
                {{- end}}
                key: neo4j-password
          {{- end }}      
```

Notably, you can't just specify a secret.  You have to specify as well what env var you're trying to expose, and what key in which secret that comes from.

Is your use case that you have a single key secret and want that as a single env var, or something else?

**Comment by alesten** [comment]
I have created a rabbitMQ service using their helm chart, during that it created a secret for me that contains the password. So for this I don't have control over what the key name is inside the secret.

I would like to specify the secret, the key within the secret and the env var i'm going to expose to.
So currently I have a single key secret to a single env var, but I could see a use case where it would be multiple secrets to multiple env vars.

**Comment by moxious** [comment]
@alesten so, thinking about how you'd want to specify this, we need to be careful; probably you'd want an array of secrets, because you might have more than one.  And order would matter, as in the configuration, the last thing overrides the first, and so on.

So some questions that go with this would be:

- Which comes first, secrets or configmap?
- Should we allow specifically secrets, or should we make the *entire environment* of the pod over-rideable rather than doing secrets vs. configmaps? 

I'm thinking it might be best not to do this with secrets per se, but allow the user to tack on as many extra env entries as they like.  In your case, since you need to specify the secret and key anyway, it might look something like this:

```
extraEnv:
  - name: MY_SECRET
     valueFrom: 
         secretKeyRef:
             name: whatever-secret
             key: supersecret
```

Noting that you could have as many of those as you wanted.   Still thinking on how best to design this to make it extensible & general.

**Comment by alesten** [comment]
@moxious I have seen in other chart that pretty much let's you use k8s syntax to do this.
This is an example from the elasticsearch chart:
https://github.com/elastic/helm-charts/blob/a56e702820740b8cdbe763b1abef9c404aeb2ecb/elasticsearch/templates/statefulset.yaml#L199

From what I can see, this will allow use of both secrets and configmaps in the same config, something like this:
```yaml
envFrom:
    - name: DEBUG_LEVEL
        valueFrom:
            configMapKeyRef:
                name: overall-config
                key: OUR_DEBUG_LEVEL
    - name: SECRET_PASSWORD
        valueFrom:
            secretKeyRef:
                name: external-system-config
                key: SUPER_SECRET_PASSWORD
```

I don't know if this would be a way to go, as the control will be a bit less when the user can write pure yaml in the config

**Comment by moxious** [comment]
PR in progress https://github.com/neo4j-contrib/neo4j-helm/pull/102

**Comment by moxious** [comment]
Fix is in version 4.1.1-4
